test/e2e/framework/service/affinity_checker.go

```diff
@@ -17,9 +17,68 @@ limitations under the License.
 package service
 
 import (
+	"fmt"
+	"net"
+	"strconv"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
 	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
 )
 
+// CheckAffinity function tests whether the service affinity works as expected.
+// If affinity is expected, the test will return true once affinityConfirmCount
+// number of same response observed in a row. If affinity is not expected, the
+// test will keep observe until different responses observed. The function will
+// return false only in case of unexpected errors.
+func CheckAffinity(execPod *v1.Pod, targetIP string, targetPort int, shouldHold bool) bool {
```

Comment:
So I don't think this shuffling makes sense to this package.  I know they were lumped before, but as we are breaking apart the framework they should go into logical sub-packages that make sense to vendor independently.  I'd argue that this should really be in the pod library.